src/Compilers/CSharp/Portable/Symbols/Synthesized/GeneratedNames.cs

```diff
@@ -390,39 +390,43 @@ internal static bool TryParseSynthesizedDelegateName(string name, out RefKindVec
                 return false;
             }
 
-            // The character after the prefix should be an open brace
-            if (name[DelegateNamePrefixLength] != '{')
-            {
-                return false;
-            }
-
             parameterCount = arity - (returnsVoid ? 0 : 1);
 
-            var lastBraceIndex = name.LastIndexOf('}');
-            if (lastBraceIndex < 0)
+            // If there are more than 16 type parameters then the synthesized delegate doesn't ref kinds encoded
+            // (and therefore no braces) so we use the end of the prefix instead.
```

Comment:
The number of type parameters doesn't affect whether ref kinds are encoded. Perhaps change to:
```suggestion
            // If there are no ref kinds encoded
            // (and therefore no braces), use the end of the prefix instead.
```

Reply:
Thanks, that sentence definitely implies a causal relationship that was not intended.